hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -544,6 +546,16 @@ public void afterTransactionCompletion(boolean success) {
 		}
 	}
 
+	@Internal
+	public void executePendingBulkOperationCleanUpActions() {
+		if ( !isTransactionCoordinatorShared ) {
```

Comment:
Not sure if this makes it "more correct", but AFAIU, the scenario we want to help users with is when there is no active transaction, so it might be worth doing this only when no transaction is in progress.

```suggestion
		if ( !isTransactionCoordinatorShared && !isTransactionInProgress() ) {
```

Reply:
@beikov I added the check in the `SessionImpl` call to `xecutePendingBulkOperationCleanUpAction`